### PR TITLE
Cyclic-Sort Algorithm

### PR DESCRIPTION
**Pull request by prishit55**
Complete documentation of cyclic sort algorithm with suggested leetcode problems.
Issue No.: #94

### CODE REVIEW

File: tutorials/basic-topics/sorting/cyclic-sort.md [hunk 1]
```diff
@@ -11,4 +11,112 @@ keywords:
   - cyclic sort
 ---
 
-<TutorialAuthors names="@TBC"/>
+<TutorialAuthors names="@prishit55"/>
+
+# Overview
+Cycle-Sort is an in-place, unstable-sorting algorithm.
```
Comment (wingkwong) [inline tutorials/basic-topics/sorting/cyclic-sort.md:17]: add a line break between L16 & L17.

File: tutorials/basic-topics/sorting/cyclic-sort.md [hunk 2]
```diff
@@ -11,4 +11,112 @@ keywords:
...
+* `in-place` : An in-place algorithm transforms input without using any other auxiliary data structure. As the algorithm executes the input is overwritten by the output.
+* `unstable` : Unstable sorting algorithm don't preserve the relative order of equal elements while sorting.
+
+
+Cycle sort algorithm factors the permutation to be sorted into number of cycles, which are individually rotated to give desired sorted result.
+
+## Algorithm
+Consider an array with `n` distinct elements. For any element `x` we can find the index at which it will occur in the sorted array by counting the number of elements smaller than `x` in the entire array.Now,
```
Comment (wingkwong) [inline tutorials/basic-topics/sorting/cyclic-sort.md:25]: add a line break between L24 & L25.

File: tutorials/basic-topics/sorting/cyclic-sort.md [hunk 3]
```diff
@@ -11,4 +11,112 @@ keywords:
...
+    }    
+     
+ }    
+```
+## Complexity
+|  Case      |  Time Complexity  |  Space Complexity 
+|------------|-------------------|-------------------
+|Best Case   | O(N^2)            |  O(1)
+|Average Case| O(N^2)            |  O(1)
+|Worst Case  | O(N^2)            |  O(1)
```
Comment (wingkwong) [inline tutorials/basic-topics/sorting/cyclic-sort.md:90]: Use LaTex for big o notation

File: tutorials/basic-topics/sorting/cyclic-sort.md [hunk 4]
```diff
@@ -11,4 +11,112 @@ keywords:
...
+        "leetCodeLink" : "https://leetcode.com/problems/find-all-numbers-disappeared-in-an-array/"
+        solutionLink" : "https://leetcode.com/submissions/detail/812254247/"
+    },
+    {
+        "problemName" : "645-Set Mismatch"
+        "difficulty" : "Easy"
+        "leetCodeLink" : "https://leetcode.com/problems/set-mismatch/"
+        "solutionLink" : "https://leetcode.com/submissions/detail/812256354/"
+    }
+]
```
Comment (wingkwong) [inline tutorials/basic-topics/sorting/cyclic-sort.md:121]: missing commas. broken here.

File: tutorials/basic-topics/sorting/cyclic-sort.md [hunk 5]
```diff
@@ -11,4 +11,112 @@ keywords:
...
+* if the element is at the correct position then do nothing 
+* Else, write the element to its intended position. That position must be inhabited by a different number `y` which has to be moved to its correct position. The process continues until we get an element at the original position of `x`.
+
+This completes one cycle. Repeating the cycle for every element will generate a sorted array.
+
+## C++ Implementation
+
+```cpp
+/*function to implement to cycle sort*/  
+void cycleSort(int a[], int n)
```
Comment (wingkwong) [inline tutorials/basic-topics/sorting/cyclic-sort.md:35]: This is not your work. It's from https://www.javatpoint.com/cycle-sort.

**Review by wingkwong — CHANGES_REQUESTED** [review]
1. Your changes break the site. Please run the app locally to verify instead of just writing on a markdown editor.
image

2. Don't copy content from other site.

**Comment by prishit55** [comment]
@wingkwong i've updated the documentation and . Please check

**Comment by prishit55** [comment]
@wingkwong I have updated the documentation. Please check it

**Review by wingkwong — CHANGES_REQUESTED** [review]
@prishit55 Before requesting a review, you have the responsibility to verify it first. 

Your PR is actually breaking the site. Please [run the app locally to verify first].(https://github.com/wingkwong/leetcode-the-hard-way/blob/main/CONTRIBUTING.md#quick-start).

Not gonna review until I can start the app.

**Comment by prishit55** [comment]
@wingkwong sorry for the inconvenience. I have fixed everything and the site is running .

Screenshot (54)
Screenshot (53)
Screenshot (52)

File: tutorials/basic-topics/sorting/cyclic-sort.md [hunk 6]
```diff
@@ -11,4 +11,97 @@ keywords:
...
+        "leetCodeLink" : "https://leetcode.com/problems/find-all-numbers-disappeared-in-an-array/",
+        "solutionLink" : "https://leetcode.com/submissions/detail/812254247/"
+    },
+    {
+        "problemName" : "645-Set Mismatch",
+        "difficulty" : "Easy",
+        "leetCodeLink" : "https://leetcode.com/problems/set-mismatch/",
+        "solutionLink" : "https://leetcode.com/submissions/detail/812256354/"
+    },
+]
```
Comment (wingkwong) [inline tutorials/basic-topics/sorting/cyclic-sort.md:105]: `solutionLink` means the link to solution section in the site. Not LC submission. If you can't find it, leave it blank.

File: tutorials/basic-topics/sorting/cyclic-sort.md [hunk 7]
```diff
@@ -11,4 +11,97 @@ keywords:
...
+            }
+        }
+        return duplicates;
+    }
+```
+**Time Complexity : $$O(N)$$**
+
+**Space Complexity : $$O(1)$$**
+
+![1](https://user-images.githubusercontent.com/77775000/193418041-3fe52e97-17e9-4f32-89f8-975b6a7c788c.jpg)
```
Comment (wingkwong) [inline tutorials/basic-topics/sorting/cyclic-sort.md:74]: 1. Did you make it yourself?
2. Put the diagram under Algorithm

File: tutorials/basic-topics/sorting/cyclic-sort.md [hunk 8]
```diff
@@ -11,4 +11,97 @@ keywords:
...
+        }
+        
+        for(int i = 0; i < n; i++) {
+            if(nums[i] != i + 1) {              //extract all those elements which are not present 
+                duplicates.push_back(nums[i]); //at their correct position
+            }
+        }
+        return duplicates;
+    }
+```
```
Comment (wingkwong) [inline tutorials/basic-topics/sorting/cyclic-sort.md:69]: 1. use the following format

````
<Tabs>
<TabItem value="cpp" label="C++">
<SolutionAuthor name="@yourname"/>

```cpp
// your code goes here
```

</TabItem>
</Tabs>
````

2. don't put inline comments. put it above the line.

3. include full code. not just the function. i.g.

```
class Solution {
public:
    vector<int> findDuplicates(vector<int>& nums) {
        // ...
    }
};
```

File: tutorials/basic-topics/sorting/cyclic-sort.md [hunk 9]
```diff
@@ -11,4 +11,97 @@ keywords:
...
+# Overview
+
+Cycle-Sort is an in-place, unstable-sorting algorithm.
+* `in-place` : An in-place algorithm transforms input without using any other auxiliary data structure. As the algorithm executes the input is overwritten by the output.
+* `unstable` : Unstable sorting algorithm don't preserve the relative order of equal elements while sorting.
+
+
+Cycle sort algorithm factors the permutation to be sorted into number of cycles, which are individually rotated to give desired sorted result.
+
+## Algorithm
```
Comment (wingkwong) [inline tutorials/basic-topics/sorting/cyclic-sort.md:25]: This part should follow what you make in the diagram.

**Review by wingkwong — CHANGES_REQUESTED** [review]

File: tutorials/basic-topics/sorting/cyclic-sort.md (hunk 7 above)
Comment (prishit55) [reply tutorials/basic-topics/sorting/cyclic-sort.md:74]: yes! I made it by myself. It is basically to explain the flow of the program for that particular question.

File: tutorials/basic-topics/sorting/cyclic-sort.md (hunk 9 above)
Comment (prishit55) [reply tutorials/basic-topics/sorting/cyclic-sort.md:25]: Unless the correct element reaches the starting index variable ' i ' will not get updated. That depicts one cycle this is what I am trying to show in the diagram.
I'll write this part somewhere in the explanation part.

File: tutorials/basic-topics/sorting/cyclic-sort.md (hunk 8 above)
Comment (prishit55) [reply tutorials/basic-topics/sorting/cyclic-sort.md]: Corrected!

File: tutorials/basic-topics/sorting/cyclic-sort.md (hunk 6 above)
Comment (prishit55) [reply tutorials/basic-topics/sorting/cyclic-sort.md]: left blank

**Review by wingkwong — APPROVED** [review]
Overall it looks ok. Will fine tune a bit myself.

**Pull request merged**